uat/src/main/java/com/aws/greengrass/testing/mqtt5/client/grpc/GRPCControlServer.java

```diff
@@ -18,26 +29,216 @@
 /**
  * Implementation of gRPC server handles requests of OTF.
  */
-class GRPCControlServer extends MqttAgentDiscoveryGrpc.MqttAgentDiscoveryImplBase {
+class GRPCControlServer {
+    private static final int PORT_MIN = 1;
+    private static final int PORT_MAX = 65535;
+
+    private static final int KEEPALIVE_OFF = 0;
+    private static final int KEEPALIVE_MIN = 5;
+    private static final int KEEPALIVE_MAX = 65535;
+
+    private static final int REASON_MIN = 0;
+    private static final int REASON_MAX = 255;
+
+    private static final int CONNECT_TIMEOUT_MIN = 1;
+
+
     private static final Logger logger = Logger.getLogger(GRPCControlServer.class.getName());
 
     @SuppressWarnings("PMD.UnusedPrivateField") // TODO: remove
     private final GRPCDiscoveryClient client;
     private final Server server;
     private final int boundPort;
-    @SuppressWarnings("PMD.UnusedPrivateField") // TODO: remove
     private MqttLib mqttLib;
+    private String shutdownReason;
 
 
     /**
-     * Build gRPC address string from host and port.
-     *
-     * @param host host part of gRPC address
-     * @param port port part of gRPC address
-     * @return address of gRPC service
+     * MQTT client control service implementation.
      */
-    public static String buildAddress(String host, int port) {
-        return host + ":" + port;
+    class MqttClientControlImpl extends MqttClientControlGrpc.MqttClientControlImplBase {
+
+        @Override
+        public void shutdownAgent(ShutdownRequest request, StreamObserver<Empty> responseObserver) {
+            // save reason
+            shutdownReason = request.getReason();
+
+            // log an event
+            logger.log(Level.INFO, "shutdownAgent: {0}", new Object[]{shutdownReason});
```

Comment:
OTF uses log4j2, let's use that.

Reply:
I suggess that code will not a part of OTF because it actually standalone (but controlled) MQTT client. We going to have at least 5 clients. Because clients can be written on C, C++, Python etc no visible reasons to integrate with OTF. That also allows to use clients in other test scenarios for example to test backend. I will check log4j2 usage in OTF and in auth plugin too.

Reply:
Our plugins use logback, but OTF does use log4j2, which I'd recommend for our usage here. The builtin java logger isn't very good.

Reply:
switched to log4j2 in 74bafa1ae9b89fcdc50ac6a2007308157046f2d2